frontend/templates/product/sections/CrossSellSection/index.tsx

```diff
@@ -0,0 +1,544 @@
+                     top={{
+                        base: 0,
+                        md: 4,
+                     }}
+                  >
+                     <FontAwesomeIcon
+                        icon={faCircleCheck}
+                        color={
+                           isSelected
+                              ? theme.colors.brand[500]
+                              : theme.colors.gray[300]
+                        }
+                        style={{
+                           width: '24px',
+                           height: '24px',
+                           transition: 'color 300ms',
+                        }}
+                     />
+                  </Box>
+               </Flex>
+               <Pricing
+                  price={formatShopifyPrice(variant.price)}
+                  compareAtPrice={
+                     variant.compareAtPrice
+                        ? formatShopifyPrice(variant.compareAtPrice)
+                        : undefined
+                  }
+                  alignSelf="flex-end"
+               />
+            </Flex>
+         </Flex>
+      </Card>
+   );
+}
+
+export interface CardImageProps {
+   src: string | null;
+   alt?: string;
+}
+
+export const CardImage = ({ src, alt }: CardImageProps) => {
+   const theme = useTheme();
+   if (src == null) {
+      return (
+         <AspectRatio ratio={1} flexGrow={0} flexShrink={0} position="relative">
+            <Box bgColor="gray.100" borderRadius="md">
+               <Circle bgColor="gray.200" size="72px">
+                  <FontAwesomeIcon
+                     icon={faImage}
+                     color={theme.colors.gray[500]}
+                     style={{
+                        width: '32px',
+                        height: '32px',
+                        transition: 'color 300ms',
+                     }}
+                  />
+               </Circle>
+            </Box>
+         </AspectRatio>
+      );
+   }
+   return (
+      <AspectRatio
+         ratio={1}
+         flexGrow={0}
+         flexShrink={0}
+         position="relative"
+         w={{
+            base: '72px',
+            md: 'unset',
+         }}
+         borderWidth={{
+            base: '1px',
+            md: '0',
+         }}
+         borderColor="gray.300"
+         borderRadius={{
+            base: 'md',
+            md: 'unset',
+         }}
+         overflow="hidden"
+      >
+         <IfixitImage
+            sizes="(max-width: 629px) 250px, (max-width: 767px) 400px, (max-width: 895px) 250px, (max-width: 1000px) 400px, 250px"
+            layout="fill"
+            objectFit="contain"
+            src={src}
+            alt={alt}
+         />
+      </AspectRatio>
+   );
+};
+
+export type PricingProps = StackProps & {
+   price: string;
+   compareAtPrice?: string | null;
+};
+
+export const Pricing = ({
+   price,
+   compareAtPrice,
+   ...stackProps
+}: PricingProps) => {
+   const isDiscounted = compareAtPrice != null;
```

Comment:
I think we should check that `price !== compareAtPrice`, in case we made a data mistake on the backend. (Or maybe we've already checked this elsewhere on the frontend, and coerced `compareAtPrice` to null?)

Reply:
That's a really good observation 👍 I think we don't need to address this now, because I have another branch with a new reusable pricing component that I made with Federico. This new component will be reusable across many places and will make sure that all the edge cases about prices will be handled consistently